### DISCUSSION TITLE
Persist no store data?

### DISCUSSION THREAD

**Discussion by Toshinaki** [pmndrs/zustand#1891] [Q&A]
I was reading the docs and noticed a no store approach to make the store update function available outside hooks.
I'd like to read saved configs from localstorage, at the very beginning when the app first render.
Does this approach also work with persist?

Or maybe even created as a hook, it will only read from the localstorage for only once?

**Comment by dbritto-dev** [comment]
@Toshinaki actually you can `read` and `write` on your store outside of `react`. Let's assume you have something like this:

```ts
import { create } from 'zustand'

type CounterStore = {
  count: number
}

const useCounterStore = create<CounterStore>()(() => ({
  count: 0
})

useCounterStore.setState({ count: 1 })
```
